test/access-token.test.js

```diff
@@ -174,30 +173,83 @@ describe('loopback.token(options)', function() {
           assert(!err);
-          assert.deepEqual(res.body, {userId: userId, state: 1});
+          assert.deepEqual(res.body, { userId: userId, state: 1 });
           done();
         });
     });
 
-  it('should skip when req.token is already present', function(done) {
-    var tokenStub = { id: 'stub id' };
-    app.use(function(req, res, next) {
-      req.accessToken = tokenStub;
-      next();
+  describe('loading multiple instances of token middleware', function() {
+    it('should skip when req.token is already present and no further options are set', function(done) {
+      var tokenStub = { id: 'stub id' };
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({ model: Token }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
+      });
+
+      request(app).get('/')
+        .set('Authorization', this.token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body).to.eql(tokenStub);
+          done();
+        });
     });
-    app.use(loopback.token({ model: Token }));
-    app.get('/', function(req, res, next) {
-      res.send(req.accessToken);
+
+    it('should not overwrite valid existing token (has "id" property) when overwriteExistingToken is falsy', function(done) {
+      var tokenStub = { id: 'stub id' };
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({
+        model: Token,
+        enableDoublecheck: true,
+      }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
+      });
+
+      request(app).get('/')
+        .set('Authorization', this.token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body).to.eql(tokenStub);
+          done();
+        });
     });
 
-    request(app).get('/')
-      .set('Authorization', this.token.id)
-      .expect(200)
-      .end(function(err, res) {
-        if (err) return done(err);
-        expect(res.body).to.eql(tokenStub);
-        done();
+    it('should overwrite existing token when enableDoublecheck and overwriteExistingToken options are truthy', function(done) {
+      var token = this.token;
+      var tokenStub = { id: 'stub id' };
+
+      app.use(function(req, res, next) {
+        req.accessToken = tokenStub;
+        next();
+      });
+      app.use(loopback.token({
+        model: Token,
+        enableDoublecheck: true,
+        overwriteExistingToken: true
+      }));
+      app.get('/', function(req, res, next) {
+        res.send(req.accessToken);
       });
+
+      request(app).get('/')
+        .set('Authorization', token.id)
+        .expect(200)
+        .end(function(err, res) {
+          if (err) return done(err);
+          expect(res.body.userId).to.eql(token.userId);
```

Comment:
because `token` is a full token object including all the properties starting with `__` (e.g. `__dataSource`, `__persisted`).
`res.body` on the other hand is a `JSON.stringify`ed version of this token object and doesn't have those properties.

when `expect(res.body).to.eql(token), you get this:
```
Uncaught AssertionError: expected { Object (id, ttl, ...) } to deeply equal { Object (__cachedRelations, __data, ...) }
  + expected - actual

   {
  -  "created": "2016-03-29T07:27:27.166Z"
  +  "__cachedRelations": {}
  +  "__data": {
  +    "created": [Date: 2016-03-29T07:27:27.166Z]
  +    "id": "BZTUdV1JCTLEnQVtUDkuM4kjZZhiq5igzeHvXi2M9ap69tt4oTYA4EdMZZ7Ay9xP"
  +    "ttl": 1209600
  +    "userId": 123
  +  }
  +  "__dataSource": [undefined]
  +  "__persisted": true
  +  "__strict": false
     "id": "BZTUdV1JCTLEnQVtUDkuM4kjZZhiq5igzeHvXi2M9ap69tt4oTYA4EdMZZ7Ay9xP"
  -  "ttl": 1209600
  -  "userId": 123
   }
```